subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022 René Beckmann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_sn_transport_udp.c
+ *
+ * @brief MQTT-SN UDP transport.
+ */
+
+#include <zephyr/net/mqtt_sn.h>
+
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_mqtt_sn, CONFIG_MQTT_SN_LOG_LEVEL);
+
+static int tp_udp_init(struct mqtt_sn_transport *transport)
+{
+	struct mqtt_sn_transport_udp *udp = UDP_CONTAINER(transport);
+	int err;
+
+	udp->sock = zsock_socket(PF_INET, SOCK_DGRAM, 0);
```

Comment:
There should also be some function that closes the socket. In the regular `MQTT` implementation the approach is a bit different, the socket is created on "connect" and closed on "disconnect" calls, but it's a bit different due to TCP (the socket is no longer usable after TCP disconnect). Perhaps we need some kind of `deinit()` function here?